relay-server/src/actors/project_buffer.rs

```diff
@@ -392,7 +401,25 @@ impl BufferService {
     fn extract_envelope(&self, row: SqliteRow) -> Result<ManagedEnvelope, BufferError> {
         let envelope_row: Vec<u8> = row.try_get("envelope")?;
         let envelope_bytes = bytes::Bytes::from(envelope_row);
-        let envelope = Envelope::parse_bytes(envelope_bytes)?;
+        let mut envelope = Envelope::parse_bytes(envelope_bytes)?;
+
+        let received_at: i64 = row.try_get("received_at")?;
+        let received_at = Duration::from_millis(received_at as u64);
+        let elapsed = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap_or_default()
+            .checked_sub(received_at)
+            .unwrap_or_default();
+
+        let start_time = Instant::now()
+            // Subtract the elapsed time (the time the envelope was kept in the spool) from the current
+            // Instant to get the timestamp when the request started.
+            .checked_sub(elapsed)
+            // If we fail to get the instant from the timestamp, we fallback to `now()`.
+            .unwrap_or_else(Instant::now);
+
+        envelope.set_start_time(start_time);
```

Comment:
If I understand this math correctly, `received_at` and `start_time` is the same, and the calculations on `elapsed` don't perform any changes. The only difference may be the time it takes from `SystemTime::now` to `Instant::now`, so a few nanoseconds? This results on the same thing:

```suggestion
        let received_at = Duration::from_millis(received_at as u64);
        envelope.set_start_time(received_at);
```

Reply:
Calculate the `received_at` to set the `start_time` is the entire point of it - in order to restore the time when the request started. Exactly what the docs say.